pkg/client/client.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+)
+
+// ObjectKey identifies a Kubernetes Object.
+type ObjectKey types.NamespacedName
+
+// Client uses methods defined in runtime.Object to determine the REST endpoint
+// for Kubernetes resources.
+
+// Interface defines the capability for a Kubernetes client.
+type Interface interface {
+	// Create saves the object obj in the Kubernetes cluster.
+	Create(ctx context.Context, obj runtime.Object) error
+
+	// Get retrieves an obj for the given object key from the Kubernetes Cluster.
+	// obj must be a struct pointer so that obj can be updated with the response
+	// returned by the Server.
+	Get(ctx context.Context, key ObjectKey, obj runtime.Object) error
+
+	// Update updates the given obj in the Kubernetes cluster. obj must be a
+	// struct pointer so that obj can be updated with the content returned by the Server.
+	Update(ctx context.Context, obj runtime.Object) error
+
+	// Delete deletes the given obj from Kubernetes cluster.
+	Delete(ctx context.Context, obj runtime.Object) error
```

Comment:
That's sort of inconsistent with `Get`?

Reply:
I am sort of struggling with it. Taking just `key` is not enough, because we need gvk info for deleting the object and `obj` helps in giving us that info. And taking both `key` and `obj` is worse than taking just the `obj`